database/dbapi.py

```diff
@@ -19,6 +19,7 @@ class Connector:
             def __init__(self, as_user):
                 usr, pwd, hst, dab = self._read_config_info(config)
                 self._establish_connection(usr, pwd, hst, dab)
+                self.client = ""
```

Comment:
I'm not 100% sure why this has been added. Is it in case `self.client` doesn't get set in the `for` loop on Line 30? If so, we need to make sure that, if the client isn't set securely in that line, the program is exited. I've made that change in my commit. Check that, please :)